src/k8s/pkg/k8sd/controllers/csrsigning/reconcile_approve_test.go

```diff
@@ -0,0 +1,123 @@
+package csrsigning
+
+import (
+	"context"
+	"crypto/rsa"
+	"errors"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	certv1 "k8s.io/api/certificates/v1"
+	v1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	k8smock "github.com/canonical/k8s/pkg/client/k8s/mock"
+	"github.com/canonical/k8s/pkg/log"
+)
+
+func TestAutoApprove(t *testing.T) {
+	for _, tc := range []struct {
+		name        string
+		csr         certv1.CertificateSigningRequest
+		validateCSR func(obj *certv1.CertificateSigningRequest, priv *rsa.PrivateKey) error
+		updateErr   error
+
+		expCtrl      ctrl.Result
+		expErr       error
+		expCondition certv1.CertificateSigningRequestCondition
+	}{
+		{
+			name: "InvalidCSR--UpdateSuccessful",
```

Comment:
Test names are split with `/`

```suggestion
			name: "InvalidCSR/UpdateSuccessful",
```